src/main/java/seedu/weme/logic/commands/EditCommand.java

```diff
@@ -85,11 +85,11 @@ public CommandResult execute(Model model) throws CommandException {
     private static Meme createEditedMeme(Meme memeToEdit, EditMemeDescriptor editMemeDescriptor) {
         assert memeToEdit != null;
 
-        Name updatedName = editMemeDescriptor.getName().orElse(memeToEdit.getName());
+        ImageUrl updatedUrl = editMemeDescriptor.getUrl().orElse(memeToEdit.getUrl());
```

Comment:
For the time being no to reduce complications. But I think it might be a protected feature in the sense that the user shouldn't anyhow change URL, but they are allowed to if the URL is corrupt or so. This is to avoid recreating the Meme object just because URL is corrupted.

Reply:
URL should not be exposed to the user at all IMO. They can delete the meme if it appears to be corrupt

Reply:
The user can key in the URL i thought. Is import supposed to abstract away this part (let's say the user just need to dump the file somewhere and do import)? @LivingOnEdge

Reply:
local url directory in application cannot be interacted with. However, import url is different as it is url outside of application. The user has to key that in.

Reply:
The user keys in a "path", they do not know it is implemented with the `URL` class.